src/lib/fountain/FountainMetadata.h

```diff
@@ -6,6 +6,14 @@ class FountainMetadata
 public:
 	static const unsigned md_size = 4;
 
+	static void to_uint8_arr(uint8_t encode_id, unsigned size, uint8_t* arr)
+	{
+		arr[0] = (encode_id & 0x7F) | ((size >> 17) & 0x80);
```

Comment:
We steal the first bit of the FountainMetadata for an extra bit of size, allowing us to address 2^25 (~33MB). This seems like a reasonable limit to me.

(notably, the fountain code implementation I'm using caps at `2^16` blocks...)